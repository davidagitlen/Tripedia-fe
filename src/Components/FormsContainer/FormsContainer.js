import React, { useState, useContext } from "react";
import "./FormsContainer.scss";
import StartForm from "../StartForm/StartForm";
import AttractionsForm from "../AttractionsForm/AttractionsForm";
import AccommodationsForm from "../AccommodationsForm/AccommodationsForm";
import FoodForm from "../FoodForm/FoodForm";
import DrinksForm from "../DrinksForm/DrinksForm";
import ServicesForm from "../ServicesForm/ServicesForm";
import SubmitTripForm from "../SubmitTripForm/SubmitTripForm";
import { FormContext } from '../../Contexts/FormContext';

export const FormsContainer = () => {
  const { formState } = useContext(FormContext);
  const [collapsed, collapseFormContainer] = useState(false);
  const [openForm, collapseForm] = useState({
    StartForm: true,
    AttractionsForm: false,
    AccommodationsForm: false,
    FoodForm: false,
    DrinksForm: false,
    ServicesForm: false,
    SubmitTripForm: false
  });
  const defaultOpenForm = {
    StartForm: false,
    AttractionsForm: false,
    AccommodationsForm: false,
    FoodForm: false,
    DrinksForm: false,
    ServicesForm: false,
    SubmitTripForm: false
  };
  const { accommodations, attractions, food, drinks, services } = formState; 
  const forms = [ accommodations, attractions, food, drinks, services ];
  const collapsedForm = collapsed ? 
  'collapsed-forms__button' :
  'not-collapsed-forms__button';
  const collapsedClass = collapsed ? 'collapsed-forms__container' : 'forms-container__container';

  return (
    <div className="wrapper-div">
      <div className={collapsedForm}>
        <div className={collapsedClass}>
          <StartForm
            collapseForm={collapseForm}
            openForm={openForm}
            defaultForm={defaultOpenForm}
          />
          <AttractionsForm
            formObject={attractions}
            collapseForm={collapseForm}
            openForm={openForm}
            defaultForm={defaultOpenForm}
          />
          <AccommodationsForm
            formObject={accommodations}
            collapseForm={collapseForm}
            openForm={openForm}
            defaultForm={defaultOpenForm}
          />
          <FoodForm
            formObject={food}
            collapseForm={collapseForm}
            openForm={openForm}
            defaultForm={defaultOpenForm}
          />
          <DrinksForm
            formObject={drinks}
            collapseForm={collapseForm}
            openForm={openForm}
            defaultForm={defaultOpenForm}
          />
          <ServicesForm
            formObject={services}
            collapseForm={collapseForm}
            openForm={openForm}
            defaultForm={defaultOpenForm}
          />
          <SubmitTripForm
            collapseForm={collapseForm}
            openForm={openForm}
            defaultForm={defaultOpenForm}
          />
        </div>
      <button
        className="collapse_button"
        onClick={e => {
          collapseFormContainer(!collapsed);
        }}
      >
        {"||"}
      </button>
      </div>
    </div>
  );
}

export default FormsContainer;
