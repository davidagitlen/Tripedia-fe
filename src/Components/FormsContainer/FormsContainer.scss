.collapsed-forms__button {
  display: flex;
  justify-content: flex-end;
  align-items: center;
  height: 80vh;
  background-color: rgba(228, 226, 226, 0);
  z-index: 3;
  position: absolute;
}
.collapse_button {
  height: 350px;
  width: 20px;
  border-radius: 0px 150px 150px 0px;
  color: lightgrey;
  background-color: rgba(245, 245, 245, 0.8);
  border: solid lightgrey 1px;
  border-left: none;
}

.collapse_button:hover {
  height: 350px;
  width: 20px;
  border-radius: 0px 150px 150px 0px;
  color: black;
  border: solid black 1px;
  border-left: none;
}
.not-collapsed-forms__button {
  display: flex;
  justify-content: flex-end;
  align-items: center;
  width: 30vw;
  height: 80vh;
  background: rgba(228, 226, 226, 0);
  z-index: 3;
  position: absolute;

  .forms-container__container {
    width: 30vw;
    height: 80vh;
    .form__container {
      width: 100%;
      height: 32vh;
      background: whitesmoke;
      display: flex;
      align-items: center;
      justify-content: center;
      flex-direction: column;
      border-top: 2px solid rgba(0, 28, 58, 0.5);
      .submit-trip__button {
        width: 75%;
        border-radius: 40px;
        height: 40px;
        font-weight: bold;
        border: 2px solid #001c3a;
        &:hover {
          background-color: #001c3a;
          color: whitesmoke;
        }
      }
      .checkbox__container {
        display: flex;
        flex-flow: row wrap;
        height: 90%;
        padding: 20px;
      }
      p {
        font-size: 1.5em;
        font-weight: bold;
        color: rgb(0, 28, 58);
      }
      .individual-checkbox__container {
        display: flex;
        width: 50%;
        padding: 10px;

        label {
          font-size: 1.1em;
          font-weight: bold;
          color: rgb(0, 28, 58);
        }
        .checkbox {
          margin-right: 10px;
        }
      }
    }

    .form-closed__container {
      width: 100%;
      height: 8vh;
      background-color: whitesmoke;
      border-top: 2px solid rgba(0, 28, 58, 0.5);
      // border-bottom: 2px solid rgba(0, 28, 58, 0.5);
      display: flex;
      flex-direction: column;
      align-items: center;
      justify-content: space-evenly;

      p {
        font-size: 1.3em;
        font-weight: bold;
        color: rgb(0, 28, 58);
      }
    }
  }
}
