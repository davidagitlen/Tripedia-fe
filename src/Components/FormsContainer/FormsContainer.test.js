import React from 'react';
import { shallow } from 'enzyme';
import FormsContainer from './FormsContainer';

describe('FormsContainer', () => {

 it("should match the wrapper with data passed in", () => {
   const wrapper = shallow(<FormsContainer />);
   expect(wrapper).toMatchSnapshot();
 });
});