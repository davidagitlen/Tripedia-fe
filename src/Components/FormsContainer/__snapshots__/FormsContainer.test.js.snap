// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormsContainer should match the wrapper with data passed in 1`] = `
<div
  className="not-collapsed-forms__button"
>
  <div
    className="forms-container__container"
  >
    <StartForm
      collapseForm={[Function]}
      defaultForm={
        Object {
          "AccommodationsForm": false,
          "AttractionsForm": false,
          "DrinksForm": false,
          "FoodForm": false,
          "ServicesForm": false,
          "StartForm": false,
          "SubmitTripForm": false,
        }
      }
      openForm={
        Object {
          "AccommodationsForm": false,
          "AttractionsForm": false,
          "DrinksForm": false,
          "FoodForm": false,
          "ServicesForm": false,
          "StartForm": true,
          "SubmitTripForm": false,
        }
      }
    />
    <AttractionsForm
      collapseForm={[Function]}
      defaultForm={
        Object {
          "AccommodationsForm": false,
          "AttractionsForm": false,
          "DrinksForm": false,
          "FoodForm": false,
          "ServicesForm": false,
          "StartForm": false,
          "SubmitTripForm": false,
        }
      }
      openForm={
        Object {
          "AccommodationsForm": false,
          "AttractionsForm": false,
          "DrinksForm": false,
          "FoodForm": false,
          "ServicesForm": false,
          "StartForm": true,
          "SubmitTripForm": false,
        }
      }
    />
    <AccommodationsForm
      collapseForm={[Function]}
      defaultForm={
        Object {
          "AccommodationsForm": false,
          "AttractionsForm": false,
          "DrinksForm": false,
          "FoodForm": false,
          "ServicesForm": false,
          "StartForm": false,
          "SubmitTripForm": false,
        }
      }
      openForm={
        Object {
          "AccommodationsForm": false,
          "AttractionsForm": false,
          "DrinksForm": false,
          "FoodForm": false,
          "ServicesForm": false,
          "StartForm": true,
          "SubmitTripForm": false,
        }
      }
    />
    <FoodForm
      collapseForm={[Function]}
      defaultForm={
        Object {
          "AccommodationsForm": false,
          "AttractionsForm": false,
          "DrinksForm": false,
          "FoodForm": false,
          "ServicesForm": false,
          "StartForm": false,
          "SubmitTripForm": false,
        }
      }
      openForm={
        Object {
          "AccommodationsForm": false,
          "AttractionsForm": false,
          "DrinksForm": false,
          "FoodForm": false,
          "ServicesForm": false,
          "StartForm": true,
          "SubmitTripForm": false,
        }
      }
    />
    <DrinksForm
      collapseForm={[Function]}
      defaultForm={
        Object {
          "AccommodationsForm": false,
          "AttractionsForm": false,
          "DrinksForm": false,
          "FoodForm": false,
          "ServicesForm": false,
          "StartForm": false,
          "SubmitTripForm": false,
        }
      }
      openForm={
        Object {
          "AccommodationsForm": false,
          "AttractionsForm": false,
          "DrinksForm": false,
          "FoodForm": false,
          "ServicesForm": false,
          "StartForm": true,
          "SubmitTripForm": false,
        }
      }
    />
    <ServicesForm
      collapseForm={[Function]}
      defaultForm={
        Object {
          "AccommodationsForm": false,
          "AttractionsForm": false,
          "DrinksForm": false,
          "FoodForm": false,
          "ServicesForm": false,
          "StartForm": false,
          "SubmitTripForm": false,
        }
      }
      openForm={
        Object {
          "AccommodationsForm": false,
          "AttractionsForm": false,
          "DrinksForm": false,
          "FoodForm": false,
          "ServicesForm": false,
          "StartForm": true,
          "SubmitTripForm": false,
        }
      }
    />
    <SubmitTripForm
      collapseForm={[Function]}
      defaultForm={
        Object {
          "AccommodationsForm": false,
          "AttractionsForm": false,
          "DrinksForm": false,
          "FoodForm": false,
          "ServicesForm": false,
          "StartForm": false,
          "SubmitTripForm": false,
        }
      }
      openForm={
        Object {
          "AccommodationsForm": false,
          "AttractionsForm": false,
          "DrinksForm": false,
          "FoodForm": false,
          "ServicesForm": false,
          "StartForm": true,
          "SubmitTripForm": false,
        }
      }
    />
  </div>
  <button
    className="collapse_button"
    onClick={[Function]}
  >
    ||
  </button>
</div>
`;
